crates/utils/src/config.rs

```diff
@@ -41,7 +52,10 @@ impl ToSocketAddrs for Endpoint {
 
 impl Display for Endpoint {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        f.write_fmt(format_args!("http://{}:{}", self.host, self.port))
+        match self.protocol {
+            Protocol::Http => f.write_fmt(format_args!("http://{}:{}", self.host, self.port)),
+            Protocol::Https => f.write_fmt(format_args!("https://{}:{}", self.host, self.port)),
+        }
```

Comment:
This works, but I think we improve readibility by implementing `Display for Protocol` and then do something like:

```suggestion
        let Endpoint { protocol, host, port } = self;
        f.write_fmt(format_args!("{protocol}://{host}:{port}"))
```